src/system.rs

```diff
@@ -296,3 +269,135 @@ impl<I: Send, E: Send> Handler<Execute<I, E>> for SystemArbiter {
         msg.exec()
     }
 }
+
+/// Builder struct for a System actor.
+///
+/// Either use `Builder::build` to create a system and start actors.
+/// Alternatively, use `Builder::run` to start the tokio runtime and
+/// run a function in its context.
+///
+/// # Examples
+///
+/// ```rust
+/// extern crate actix;
+///
+/// use actix::prelude::*;
+/// use actix::clock::Clock;
+/// use std::time::Duration;
+///
+/// struct Timer {
+///     dur: Duration,
+/// }
+///
+/// impl Actor for Timer {
+///     type Context = Context<Self>;
+///
+///     // stop system after `self.dur` seconds
+///     fn started(&mut self, ctx: &mut Context<Self>) {
+///         ctx.run_later(self.dur, |act, ctx| {
+///             // Stop current running system.
+///             System::current().stop();
+///         });
+///     }
+/// }
+///
+/// fn main() {
+///     // create a custom clock instance
+///     let clock = Clock::new();
+///
+///     // initialize system and run it
+///     // This function blocks current thread
+///     let code = System::builder().clock(clock).run(|| {
+///         // Start `Timer` actor. It will use the provided clock instance.
+///         Timer {
+///             dur: Duration::new(0, 1),
+///         }.start();
+///     });
+///
+///     std::process::exit(code);
+/// }
+/// ```
+pub struct Builder {
+    /// Name of the System. Defaults to "actix" if unset.
+    name: Cow<'static, str>,
+
+    /// Tokio runtime builder.
+    runtime: RuntimeBuilder,
+}
+
+impl Builder {
+    fn new() -> Self {
+        Builder {
+            name: Cow::Borrowed("actix"),
+            runtime: RuntimeBuilder::new(),
+        }
+    }
+
+    /// Sets the name of the System.
+    pub fn name<T: Into<String>>(mut self, name: T) -> Self {
+        self.name = Cow::Owned(name.into());
+        self
+    }
+
+    /// Set the Clock instance that will be used by this System.
+    ///
+    /// Defaults to the system clock.
+    pub fn clock(mut self, clock: Clock) -> Self {
+        self.runtime.clock(clock);
+        self
+    }
+
+    /// Create new System.
+    ///
+    /// This method panics if it can not create tokio runtime
+    pub fn build(self) -> SystemRunner {
```

Comment:
Just to confirm, it is rather unusual to end a builder with anything but `build()`. Besides, this would make the interface different from the Arbiter builder.